packages/starknet-snap/src/utils/starknetUtils.test.ts

```diff
@@ -0,0 +1,124 @@
+import type { EstimateFee, Invocations } from 'starknet';
+import { constants, TransactionType } from 'starknet';
+
+import { mockAccount, prepareMockAccount } from '../rpcs/__tests__/helper';
+import { FeeTokenUnit } from '../types/snapApi';
+import type { SnapState } from '../types/snapState';
+import { STARKNET_SEPOLIA_TESTNET_NETWORK } from './constants';
+import * as starknetUtils from './starknetUtils';
+
+const invocations: Invocations = [
+  {
+    type: TransactionType.INVOKE,
+    payload: {
+      contractAddress:
+        '0x00b28a089e7fb83debee4607b6334d687918644796b47d9e9e38ea8213833137',
+      entrypoint: 'functionName',
+      calldata: ['1', '1'],
+    },
+  },
+];
+describe('getEstimatedFees', () => {
+  const createMockSpy = async (
+    state: SnapState,
+    estimateFeeResp: EstimateFee,
+    deployed: boolean,
+  ) => {
+    const chainId = constants.StarknetChainId.SN_SEPOLIA;
+    const account = await mockAccount(chainId);
+    prepareMockAccount(account, state);
+    const accountDeployedSpy = jest.spyOn(starknetUtils, 'isAccountDeployed');
+    accountDeployedSpy.mockResolvedValue(deployed);
+    const estimateBulkFeeSpy = jest.spyOn(starknetUtils, 'estimateFeeBulk');
+    estimateBulkFeeSpy.mockResolvedValue([estimateFeeResp]);
```

Comment:
estimateFeeResp seems appear everywhere, may be you can put  estimateFeeResp into here, and override it unless u need to, and you can also return estimateFeeResp in that func